src/main/java/com/taxi/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,200 @@
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create new Car "
+                    + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        //сделать прватный метод, который также будет добавлять драйверов в БД
+        Car car = new Car();
+        String getStatement = "SELECT * FROM cars c "
+                + " JOIN manufacturers m on c.manufacturer_id = m.id"
+                + " WHERE c.id = ? AND c.isDeleted = false";
+        try (Connection connection = connectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getStatement)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get Car with id "
+                    + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> carList = new ArrayList<>();
+        String getAllStatement = "SELECT * FROM cars WHERE isDeleted = false ";
+        try (Connection connection = connectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(getAllStatement)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCar(resultSet));
+            }
+            return carList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all Cars to the List !"
+                    + carList.toString(), e);
+        }
+    }
+
+    @Transactional
+    @Override
+    public Car update(Car car) {
+        String updateStatement = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND isDeleted = false ;";
+        try (Connection connection = connectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(updateStatement)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            removeDriversForCar(car);
+            car.setDriverSet(updateDriversForCar(car));
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update Car "
+                    + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET isDeleted = true "
+                + "WHERE id = ? AND isDeleted = false;";
```

Comment:
Не вижу смысла второго условия
```suggestion
        String query = "UPDATE cars SET isDeleted = true "
                + "WHERE id = ?;";
```